### PR TITLE
Rules Command

### PR DESCRIPTION
**Pull request by EdgarSaldivar**

### CODE REVIEW

File: src/bot/commands/rules.command.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { InjectRepository } from '@mikro-orm/nestjs';
+import { EntityRepository } from '@mikro-orm/postgresql';
+
+import { RulesService } from '../rules.service';
+import { TimingLogInterceptor } from '../interceptors/logging.interceptor';
+import { ValidationErrorFilter } from '../filters/validation-error.filter';
+import { CatchallErrorFilter } from '../filters/catchall-error.filter';
+import { GuildEntity } from 'src/entities/guild/guild.entity';
+
+export class CheckScoreDTO {}
```
Comment (dudeofawesome) [inline src/bot/commands/rules.command.ts]: Do we need this DTO?

File: src/bot/commands/rules.command.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+@Command({
+  name: 'rules',
+  description: `Rules for Virginity Bot on your server.`,
+  defaultMemberPermissions: PermissionFlagsBits.SendMessages,
+  dmPermission: false,
+})
+@Injectable()
+@UseFilters(ValidationErrorFilter, CatchallErrorFilter)
```
Comment (dudeofawesome) [inline src/bot/commands/rules.command.ts]: Are there any parameters to validate?

File: src/bot/commands/rules.command.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+@UseInterceptors(TimingLogInterceptor)
+export class RulesCommand {
+  private readonly logger = new Logger(RulesCommand.name);
+
+  constructor(
+    private readonly orm: MikroORM,
+    @InjectRepository(GuildEntity)
+    private readonly guilds: EntityRepository<GuildEntity>,
+    private readonly rulesBoard: RulesService,
```
Comment (dudeofawesome) [inline src/bot/commands/rules.command.ts]: This should be called `rules` or `rules_service`

File: src/bot/commands/rules.command.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  ): Promise<Message> {
+    if (interaction.guildId == null || interaction.guild == null) {
+      this.logger.error([`interaction.guildId was null somehow`, interaction]);
+      throw new Error(`interaction.guildId was null somehow`);
+    } else if (interaction.channel == null) {
+      this.logger.error([`interaction.channel was null somehow`, interaction]);
+      throw new Error(`interaction.channel was null somehow`);
+    }
+
+    await interaction.deferReply();
```
Comment (dudeofawesome) [inline src/bot/commands/rules.command.ts]: This command should be fast enough that we don't need to defer the reply.

File: src/bot/rules.service.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import * as cronstrue from 'cronstrue';
+
+import { GuildEntity } from 'src/entities/guild/guild.entity';
+
+@Injectable()
+export class RulesService {
+  constructor(
+    private readonly orm: MikroORM,
+    @InjectRepository(GuildEntity)
+    private readonly guilds: EntityRepository<GuildEntity>,
```
Comment (dudeofawesome) [inline src/bot/rules.service.ts]: Do we ever use this?

File: src/bot/rules.service.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  ) {}
+
+  @UseRequestContext()
```
Comment (dudeofawesome) [inline src/bot/rules.service.ts]: Don't think we need to talk to the DB here?

File: src/bot/commands/rules.command.ts (hunk 4 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: It complained at somepoint about it not being deferred. But sure I'll test it again without.

File: src/bot/commands/rules.command.ts (hunk 3 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: Valid

File: src/bot/commands/rules.command.ts (hunk 1 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: nope.

File: src/bot/commands/rules.command.ts (hunk 2 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: same as above

File: src/bot/rules.service.ts (hunk 5 above)
Comment (EdgarSaldivar) [reply src/bot/rules.service.ts]: line 14?

**Comment by EdgarSaldivar** [comment]
Need your feedback on where the linter is failing btw.

File: src/bot/commands/rules.command.ts (hunk 4 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: Can confirm, event will still fail without a deferral.

File: src/bot/commands/rules.command.ts (hunk 1 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: Addressed.

File: src/bot/commands/rules.command.ts (hunk 2 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: Addressed.

File: src/bot/commands/rules.command.ts (hunk 3 above)
Comment (EdgarSaldivar) [reply src/bot/commands/rules.command.ts]: Addressed.

File: src/bot/rules.service.ts (hunk 6 above)
Comment (EdgarSaldivar) [reply src/bot/rules.service.ts]: Addressed.

File: src/bot/commands/rules.command.ts (hunk 4 above)
Comment (dudeofawesome) [reply src/bot/commands/rules.command.ts]: Did you change the response method? Or just remove the deferral. `.followUp` only works with deferred interactions.

File: src/bot/rules.service.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+import { stripIndent } from 'common-tags';
+import * as cronstrue from 'cronstrue';
+
+import { GuildEntity } from 'src/entities/guild/guild.entity';
+
+@Injectable()
+export class RulesService {
+  constructor(private readonly orm: MikroORM) {}
+
+  @UseRequestContext()
```
Comment (dudeofawesome) [inline src/bot/rules.service.ts]: We shouldn't need this anymore if we're not talking to the DB.

File: src/bot/rules.service.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import { UseRequestContext, MikroORM } from '@mikro-orm/core';
+import { EntityRepository } from '@mikro-orm/postgresql';
+import { stripIndent } from 'common-tags';
+import * as cronstrue from 'cronstrue';
+
+import { GuildEntity } from 'src/entities/guild/guild.entity';
+
+@Injectable()
+export class RulesService {
+  constructor(private readonly orm: MikroORM) {}
```
Comment (dudeofawesome) [inline src/bot/rules.service.ts]: Don't think we ever use this.

**Review by dudeofawesome — APPROVED** [review]

**Pull request merged**